src/net.md

```diff
@@ -1019,6 +1020,151 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-progress-with-range]: #ex-progress-with-range
+<a name="ex-progress-with-range"/>
+## Make a partial download with HTTP range headers
+
+[![reqwest-badge]][reqwest] [![cat-net-badge]][cat-net]
+
+Uses [`reqwest::Client::head`] to get the content-length and validate if the server sets the header
+[`reqwest::header::ContentRange`], required to confirm the support of partial downloads.
+
+If supported downloads the content using [`reqwest::get`], setting the [`reqwest::header::Range`]
+to do partial downloads in chunks of 100 bytes, between these writes basic progress messages.
+
+Range header, defined in [RFC7233][HTTP Range RFC7233].
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+
+use std::io::{Read, Write};
+use std::fs::File;
+
+use reqwest::header::{ContentRange, ContentRangeSpec, Range};
+
+# error_chain! {
+#     foreign_links {
+#         Io(std::io::Error);
+#         Reqwest(reqwest::Error);
+#     }
+# }
+
+const BUFFER_SIZE: u64 = 100;
+
+struct PartialRange {
+    content_length: u64,
+    range_ends: u64,
+    ranges: Vec<u64>,
+}
+
+impl PartialRange {
+    pub fn new(content_range: &ContentRangeSpec) -> PartialRange {
+        let (content_length, range_starts, range_ends) = match *content_range {
+            ContentRangeSpec::Bytes {
+                range: Some(range),
+                instance_length: Some(content_length),
+            } => (content_length, range.0, range.1),
+            _ => (0, 0, 0),
```

Comment:
the iterator with (0,0,0) does not make sense/ would be invalid so I would suggest returning `Err("invalid range speccification")?` in this match leg,